server/asset/dcr/dcr.go

```diff
@@ -926,53 +914,53 @@ func (dcr *Backend) getUnspentTxOut(txHash *chainhash.Hash, vout uint32) (*chain
 
 // Get information for an unspent transaction output, plus the verbose
 // transaction.
-func (dcr *Backend) getTxOutInfo(txHash *chainhash.Hash, vout uint32) (*chainjson.GetTxOutResult, *chainjson.TxRawResult, []byte, error) {
-	txOut, pkScript, err := dcr.getUnspentTxOut(txHash, vout)
+func (dcr *Backend) getTxOutInfo(ctx context.Context, txHash *chainhash.Hash, vout uint32) (*chainjson.GetTxOutResult, *chainjson.TxRawResult, []byte, error) {
+	txOut, pkScript, err := dcr.getUnspentTxOut(ctx, txHash, vout)
 	if err != nil {
 		return nil, nil, nil, err
 	}
-	verboseTx, err := dcr.node.GetRawTransactionVerbose(txHash)
+	verboseTx, err := dcr.node.GetRawTransactionVerbose(ctx, txHash)
 	if err != nil {
-		return nil, nil, nil, fmt.Errorf("GetRawTransactionVerbose for txid %s: %w", txHash, err)
+		return nil, nil, nil, fmt.Errorf("GetRawTransactionVerbose for txid %s: %w", txHash, translateRPCCancelErr(err))
 	}
 	return txOut, verboseTx, pkScript, nil
 }
 
 // Get the block information, checking the cache first. Same as
 // getDcrBlock, but takes a string argument.
-func (dcr *Backend) getBlockInfo(blockid string) (*dcrBlock, error) {
+func (dcr *Backend) getBlockInfo(ctx context.Context, blockid string) (*dcrBlock, error) {
```

Comment:
Why not use `dcr.ctx` instead of passing the context as an argument for these unexported functions throughout?

Reply:
This traces back to the Confirmations and FundingCoin methods where the caller wants to specify timeout.  Same reason for the getMainchainDcrBlock method.